admin-dev/themes/new-theme/js/pages/product/combination/bulk-form-handler.ts

```diff
@@ -0,0 +1,224 @@
+      bulkCombinationsBtn.dataset.modalConfirmLabel as string,
+      bulkCombinationsBtn.dataset.modalCancelLabel as string,
+    ));
+  }
+
+  private showFormModal(formUrl: string, confirmButtonLabel: string, closeButtonLabel: string): void {
+    const selectedCombinationsCount = this.getSelectedCheckboxes().length;
+
+    let initialSerializedData: string;
+    const iframeModal = new IframeModal({
+      id: CombinationMap.bulkFormModalId,
+      iframeUrl: formUrl,
+      autoSizeContainer: 'form[name="bulk_combination"]',
+      closable: true,
+      confirmButtonLabel: confirmButtonLabel.replace(/%combinations_number%/, String(selectedCombinationsCount)),
+      closeButtonLabel,
+      onLoaded: (iframe: HTMLIFrameElement) => {
+        // Disable submit button as long as the form data has not changed
+        iframeModal.modal.confirmButton?.setAttribute('disabled', 'disabled');
+
+        if (iframe.contentWindow) {
+          // eslint-disable-next-line max-len
+          const form: HTMLFormElement | null = iframe.contentWindow.document.querySelector<HTMLFormElement>('form[name="bulk_combination"]');
+
+          if (form) {
+            initialSerializedData = this.serializeForm(form);
+            form.addEventListener('change', () => {
+              const currentSerializedData: string = this.serializeForm(form);
+
+              if (currentSerializedData === initialSerializedData) {
+                iframeModal.modal.confirmButton?.setAttribute('disabled', 'disabled');
+              } else {
+                iframeModal.modal.confirmButton?.removeAttribute('disabled');
+              }
+            });
+          }
+        }
+      },
+      confirmCallback: () => {
+        if (iframeModal.modal.iframe.contentWindow) {
+          // eslint-disable-next-line max-len
+          const form: HTMLFormElement | null = iframeModal.modal.iframe.contentWindow.document.querySelector<HTMLFormElement>('form[name="bulk_combination"]');
+
+          if (form) {
+            this.submitForm(form);
+          }
+        }
+      },
+    });
+
+    iframeModal.show();
+  }
+
+  private serializeForm(form: HTMLFormElement): string {
+    // @ts-ignore
+    return new URLSearchParams(new FormData(form)).toString();
+  }
+
+  /**
+   * Delegated event listener on tabContainer, because every checkbox is re-rendered with dynamic pagination
+   */
+  private listenCheckboxesChange(): void {
+    this.tabContainer.addEventListener('change', (e) => {
+      if (!(e.target instanceof HTMLInputElement)) {
+        return;
+      }
+
+      if (e.target.id === CombinationMap.bulkSelectAllInPageId) {
+        this.checkAll(e.target.checked);
+      }
+      this.toggleBulkActions();
+    });
+  }
+
+  private checkAll(checked: boolean) {
+    const allCheckboxes = this.tabContainer
+      .querySelectorAll(CombinationMap.tableRow.isSelectedCombination) as NodeListOf<HTMLInputElement>;
+
+    allCheckboxes.forEach((checkbox) => {
```

Comment:
`allCheckboxes?.forEach((checkbox) => {`

Reply:
Not needed from the signature I assume a `Nodelist` is always returned but it can be empty